python/plugins/db_manager/db_plugins/postgis/connector.py

```diff
@@ -1014,7 +1014,16 @@ def moveTable(self, table, new_table, new_schema=None):
         self._commit()
 
     def createView(self, view, query):
-        sql = "CREATE VIEW %s AS %s" % (self.quoteId(view), query)
+        view_name_parts = view.split('.')
+
+        if len(view_name_parts) > 2:
+            # Raise an error when more than one period is used.
+            raise ValueError("Invalid view name: Please use the format 'schema.viewname', or enter only the viewname for the public schema.")
```

Comment:
A ValueError won't get gracefully caught here -- this should probably be:

```suggestion
            raise DbError("Invalid view name: Please use the format 'schema.viewname', or enter only the viewname for the public schema.")
```